polymorphic/query.py

```diff
@@ -25,6 +25,25 @@
 Polymorphic_QuerySet_objects_per_request = CHUNK_SIZE
 
 
+if django.VERSION >= (1, 9):
+    # We ignore this on django < 1.9, as ModelIterable didn't yet exist.
+    from django.db.models.query import ModelIterable
+
+
+    class PolymorphicModelIterable(ModelIterable):
+        def __iter__(self):
+            base_iter = super(PolymorphicModelIterable, self).__iter__()
+
+            if self.queryset.polymorphic_disabled:
+                for o in base_iter:
+                    yield o
+                return
+
+            real_instances = self.queryset._get_real_instances(base_iter)
+            for obj in real_instances:
+                yield obj
```

Comment:
> Should we be chunking with Polymorphic_QuerySet_objects_per_request?
Yes, you'll need that. Suppose you read 10+ objects from an iterator, but the actual table is 1000+ objects. Without chunking you'd retrieve all those objects, and then the iterator walks over it.